csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DotNet.cs

```diff
@@ -76,22 +115,21 @@ public bool AddPackage(string folder, string package)
 
         private IList<string> GetListed(string args, string artifact)
         {
-            progressMonitor.RunningProcess($"{dotnet} {args}");
-            var pi = MakeDotnetStartInfo(args, redirectStandardOutput: true);
-            var exitCode = pi.ReadOutput(out var artifacts);
-            if (exitCode != 0)
+            if (RunCommand(args, out var artifacts))
             {
-                progressMonitor.CommandFailed(dotnet, args, exitCode);
-                return new List<string>();
+                progressMonitor.LogInfo($"Found {artifact}s: {string.Join("\n", artifacts)}");
+                return artifacts;
             }
-            progressMonitor.LogInfo($"Found {artifact}s: {string.Join("\n", artifacts)}");
-            return artifacts;
+            return new List<string>();
         }
 
         public bool Exec(string execArgs)
         {
             var args = $"exec {execArgs}";
             return RunCommand(args);
         }
+
+        [GeneratedRegex("Restored\\s+(.+\\.csproj)", RegexOptions.Compiled)]
```

Comment:
Is the output of `dotnet restore` always English?

Reply:
Uh, that is an excellent question!
I have tried to set the CLI UI language to german and verified that it prints help in german (so it does respect setting the CLI language). The output of `dotnet restore` is still in english.
Do you have any other suggestions?

Reply:
It seems to be translated to German: https://github.com/NuGet/NuGet.Build.Localization/blob/c65073ce9be5c699d7b1cbee913223b77988f6ad/localize/DEU/15/NuGet.Commands.dll.lcl#L1459

`DOTNET_CLI_UI_LANGUAGE=de dotnet restore` prints the message in German. So maybe we need to set `DOTNET_CLI_UI_LANGUAGE=en` before running the command. Also, nuget.exe seems to have an option named `-ForceEnglishOutput`. I don't know if this could be passed to the dotnet CLI to forward it to nuget.

Reply:
You are absolutely right. The reason why I experienced that I still got the output in english was because I failed to restart VS code after setting the environment variable.
Fortunately, we can set the environment variable as a part of the .NET process start info (this only applies to the process itself and its child-processes).
I have tried to test it locally and set my `DOTNET_CLI_UI_LANGUAGE=de`. Without explicitly overriding it in the .NET process start info the optimization is not applied as the output no longer has the expected format. However, it we set  `DOTNET_CLI_UI_LANGUAGE=en` the optimization works as expected.